src/main/java/io/passport/server/controller/KeycloakController.java

```diff
@@ -0,0 +1,37 @@
+package io.passport.server.controller;
+
+import io.passport.server.model.Credentials;
+import io.passport.server.service.KeycloakService;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.representations.AccessTokenResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+/**
+ * Keycloak authorization tools and request.
+ */
+@RestController
+@RequestMapping("/user")
+public class KeycloakController {
+
+    @Autowired
+    private KeycloakService keycloakService;
+
+    /**
+     * Login request which sends the necessary credentials to Keycloak along with environment variables, and returns the authentication token.
+     * @param user User credentials.
+     * @return Token
+     */
+    @PostMapping("/login")
+    public ResponseEntity<?> login(@RequestBody Credentials user) {
+        try {
+            Keycloak keycloak = keycloakService.newKeycloakBuilderWithPasswordCredentials(user.username, user.password);
+            AccessTokenResponse tokenResponse = keycloak.tokenManager().grantToken();
+
+            return ResponseEntity.ok(tokenResponse);
+        } catch (Exception e) {
+            return ResponseEntity.status(401).body("Invalid username or password");
```

Comment:
You are masking out other Keycloak exceptions. Please sure about that try block can only throw invalid username/password exception.